cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -371,11 +373,23 @@ private static String addTLSConfigToMirrorMaker2ConnectorConfig(Map<String, Obje
         return securityProtocol;
     }
 
+    static class ConnectorsComparator implements Comparator<Map<String, Object>>, Serializable {
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        public int compare(Map<String, Object> m1, Map<String, Object> m2) {
+            String name1 = m1.get("name") == null ? "" : m1.get("name").toString();
+            String name2 = m2.get("name") == null ? "" : m2.get("name").toString();
```

Comment:
I've been enjoying the Optional class recently - these lines could be updated to:
```
String name1 = Optional.ofNullable(props1.get("name")).map(Object::toString).orElse("");
String name2 = Optional.ofNullable(props2.get("name")).map(Object::toString).orElse("");
```

Reply:
I agree with Andrew here, the use of the Optional makes the intent of the line slightly clearer. Though by convention Strimzi as a project does tend to use null checks over optionals...

Perhaps one of the other maintainers has an opinion on this, @ppatierno ?

Reply:
TBH, I'm probably more old school. The original code from @stanlyDoge is much easier to read for me. But I do not have a problem if he changes it according to your suggestion.

Reply:
I am old school as @scholzj and I agree with him that the actual code is more readable and understandable.